Added ability to reference Self on ApplicationUnderTest and BrowserWi…

icnocop opened the pull request:
…ndow

icnocop committed:
Added ability to reference Self on ApplicationUnderTest and BrowserWindow

FantasticFiasco commented on src/CUITe/Controls/UITesting/ApplicationUnderTest.cs:
```diff
@@ -0,0 +1,78 @@
+        /// <exception cref="System.NotImplementedException"></exception>
+        public override ControlBase Parent
+        {
+            get { throw new System.NotImplementedException(); }
```
Is it correct to throw `NotImplementedException`, or should you throw `NotSupportedException` or similar?

icnocop replied:
Fixed. Thank you! :)

icnocop committed:
throwing NotSupportedException instead of NotImplementedException
created separate unit test for asserting on the Self property of ApplicationUnderTest and BrowserWindow
